File: chaos-backend/check_mongodb_data.js
// MongoDB data structure analysis script
const { MongoClient } = require('mongodb');

async function checkMongoDBData() {
    const client = new MongoClient('mongodb://localhost:27017');
    
    try {
        await client.connect();
        const db = client.db('chaos_game');
        
        console.log('='.repeat(60));
        console.log('🔍 MONGODB DATA STRUCTURE ANALYSIS');
        console.log('='.repeat(60));
        
        // List all collections
        console.log('\n📋 COLLECTIONS IN DATABASE:');
        const collections = await db.listCollections().toArray();
        
        for (const collection of collections) {
            const count = await db.collection(collection.name).countDocuments();
            console.log(`  - ${collection.name}: ${count} documents`);
        }
        
        console.log('\n' + '='.repeat(60));
        
        // Check runtime_flags collection
        console.log('\n🚩 RUNTIME FLAGS COLLECTION:');
        const runtimeFlags = db.collection('runtime_flags');
        const flagsDocs = await runtimeFlags.find().toArray();
        
        for (const doc of flagsDocs) {
            console.log(`\n📄 Document ID: ${doc._id}`);
            for (const [key, value] of Object.entries(doc)) {
                if (key !== '_id') {
                    console.log(`  ${key}: ${value} (${typeof value})`);
                }
            }
        }
        
        console.log('\n' + '='.repeat(60));
        
        // Check configurations collection
        const configCollections = collections.filter(c => 
            c.name.toLowerCase().includes('config')
        );
        
        if (configCollections.length > 0) {
            console.log('\n⚙️  CONFIGURATION COLLECTIONS:');
            
            for (const collection of configCollections) {
                console.log(`\n📁 Collection: ${collection.name}`);
                const coll = db.collection(collection.name);
                const count = await coll.countDocuments();
                console.log(`  Total documents: ${count}`);
                
                if (count > 0) {
                    const sampleDocs = await coll.find().limit(3).toArray();
                    console.log(`  Sample documents:`);
                    
                    for (let i = 0; i < sampleDocs.length; i++) {
                        const doc = sampleDocs[i];
                        console.log(`\n  📄 Document ${i+1}:`);
                        console.log(`    ID: ${doc._id}`);
                        
                        for (const [key, value] of Object.entries(doc)) {
                            if (key !== '_id') {
                                if (typeof value === 'object' && value !== null) {
                                    if (Array.isArray(value)) {
                                        console.log(`    ${key}: Array with ${value.length} items`);
                                    } else {
                                        console.log(`    ${key}: Object with ${Object.keys(value).length} keys`);
                                    }
                                } else {
                                    console.log(`    ${key}: ${value} (${typeof value})`);
                                }
                            }
                        }
                    }
                }
            }
        } else {
            console.log('\n⚠️  No configuration collections found');
        }
        
        console.log('\n' + '='.repeat(60));
        
        // Summary
        console.log('\n📊 SUMMARY:');
        console.log(`  Total collections: ${collections.length}`);
        console.log(`  Runtime flags: ${flagsDocs.length} documents`);
        
        let totalConfigDocs = 0;
        for (const collection of configCollections) {
            const count = await db.collection(collection.name).countDocuments();
            totalConfigDocs += count;
        }
        
        console.log(`  Configuration documents: ${totalConfigDocs}`);
        
        console.log('\n✅ MongoDB data structure analysis completed!');
        
    } catch (error) {
        console.error('❌ Error:', error);
    } finally {
        await client.close();
    }
}

checkMongoDBData();

File: chaos-backend/check_mongodb_data.py
#!/usr/bin/env python3
"""
Script to check MongoDB data structure and organization
"""

import pymongo
from pprint import pprint
import json

def check_mongodb_data():
    """Check MongoDB data structure and organization"""
    
    # Connect to MongoDB
    client = pymongo.MongoClient("mongodb://localhost:27017")
    db = client["chaos_game"]
    
    print("=" * 60)
    print("🔍 MONGODB DATA STRUCTURE ANALYSIS")
    print("=" * 60)
    
    # List all collections
    print("\n📋 COLLECTIONS IN DATABASE:")
    collections = db.list_collection_names()
    for collection_name in collections:
        count = db[collection_name].count_documents({})
        print(f"  - {collection_name}: {count} documents")
    
    print("\n" + "=" * 60)
    
    # Check runtime_flags collection
    print("\n🚩 RUNTIME FLAGS COLLECTION:")
    runtime_flags = db["runtime_flags"]
    flags_docs = list(runtime_flags.find())
    
    for doc in flags_docs:
        print(f"\n📄 Document ID: {doc.get('_id')}")
        for key, value in doc.items():
            if key != '_id':
                print(f"  {key}: {value} ({type(value).__name__})")
    
    print("\n" + "=" * 60)
    
    # Check configurations collection (if exists)
    config_collections = [name for name in collections if 'config' in name.lower()]
    
    if config_collections:
        print(f"\n⚙️  CONFIGURATION COLLECTIONS:")
        for collection_name in config_collections:
            print(f"\n📁 Collection: {collection_name}")
            collection = db[collection_name]
            
            # Get sample documents
            sample_docs = list(collection.find().limit(5))
            
            if sample_docs:
                print(f"  Total documents: {collection.count_documents({})}")
                print(f"  Sample documents:")
                
                for i, doc in enumerate(sample_docs):
                    print(f"\n  📄 Document {i+1}:")
                    print(f"    ID: {doc.get('_id')}")
                    
                    # Show key fields
                    for key, value in doc.items():
                        if key != '_id':
                            if isinstance(value, dict):
                                print(f"    {key}: {type(value).__name__} with {len(value)} keys")
                            elif isinstance(value, list):
                                print(f"    {key}: {type(value).__name__} with {len(value)} items")
                            else:
                                print(f"    {key}: {value} ({type(value).__name__})")
            else:
                print("  No documents found")
    else:
        print("\n⚠️  No configuration collections found")
    
    print("\n" + "=" * 60)
    
    # Check for any actor-related collections
    actor_collections = [name for name in collections if 'actor' in name.lower()]
    
    if actor_collections:
        print(f"\n👤 ACTOR COLLECTIONS:")
        for collection_name in actor_collections:
            print(f"\n📁 Collection: {collection_name}")
            collection = db[collection_name]
            count = collection.count_documents({})
            print(f"  Total documents: {count}")
            
            if count > 0:
                sample = collection.find_one()
                print(f"  Sample document structure:")
                for key, value in sample.items():
                    if key != '_id':
                        print(f"    {key}: {type(value).__name__}")
    
    print("\n" + "=" * 60)
    
    # Check for any resource-related collections
    resource_collections = [name for name in collections if 'resource' in name.lower()]
    
    if resource_collections:
        print(f"\n💎 RESOURCE COLLECTIONS:")
        for collection_name in resource_collections:
            print(f"\n📁 Collection: {collection_name}")
            collection = db[collection_name]
            count = collection.count_documents({})
            print(f"  Total documents: {count}")
            
            if count > 0:
                sample = collection.find_one()
                print(f"  Sample document structure:")
                for key, value in sample.items():
                    if key != '_id':
                        print(f"    {key}: {type(value).__name__}")
    
    print("\n" + "=" * 60)
    
    # Summary
    print("\n📊 SUMMARY:")
    print(f"  Total collections: {len(collections)}")
    print(f"  Runtime flags: {len(flags_docs)} documents")
    
    total_config_docs = 0
    for collection_name in config_collections:
        total_config_docs += db[collection_name].count_documents({})
    
    print(f"  Configuration documents: {total_config_docs}")
    
    print("\n✅ MongoDB data structure analysis completed!")
    
    client.close()

if __name__ == "__main__":
    check_mongodb_data()
